### PR TITLE
UIIN-1904: Filters/facets on Subject browse form

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
## Purpose

https://issues.folio.org/browse/UIIN-1904
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by uladislausamets — APPROVED** [review]

File: src/components/InstancesList/InstancesList.js [hunk 1]
```diff
@@ -765,29 +773,31 @@ class InstancesList extends React.Component {
...
 
-    const customPaneSubTextBrowse = browseSelected ? <FormattedMessage id="ui-inventory.title.subTitle.browseCall" /> : null;
-    const searchFieldButtonLabelBrowse = browseSelected ? <FormattedMessage id="ui-inventory.browse" /> : null;
-    const titleBrowse = browseSelected ? <FormattedMessage id="ui-inventory.title.browseCall" /> : null;
-    const notLoadedMessageBrowse = browseSelected ? <FormattedMessage id="ui-inventory.notLoadedMessage.browseCall" /> : null;
+    const browseFilter = () => {
+      const { renderer } = getFilterConfig();
+      if (browseSelected === browseModeOptions.SUBJECTS) {
+        return renderer;
+      } else return renderFilters;
```
Comment (vashjs) [inline src/components/InstancesList/InstancesList.js:792]: else isn't required here, just return renderFilters

File: src/components/InstancesList/InstancesList.js [hunk 2]
```diff
@@ -167,8 +170,14 @@ class InstancesList extends React.Component {
...
-    const visibleColumns = this.state.browseSelected ? new Set([...columns]) : new Set([...columns, ...NON_TOGGLEABLE_COLUMNS]);
-    return ALL_COLUMNS.filter(key => visibleColumns.has(key));
+    const visibleColumns = Object.values(browseModeOptions).some(el => this.state.browseSelected.includes(el)) ?
+      new Set([...columns])
+      :
+      new Set([...columns, ...NON_TOGGLEABLE_COLUMNS]);
+
+    if (Object.values(browseModeOptions).some(el => this.state.browseSelected.includes(el))) {
+      return Array.from(visibleColumns);
+    } else return ALL_COLUMNS.filter(key => visibleColumns.has(key));
```
Comment (vashjs) [inline src/components/InstancesList/InstancesList.js:180]: else is'nt required

**Review by vashjs — APPROVED** [review]

File: src/components/InstancesList/InstancesList.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply src/components/InstancesList/InstancesList.js]: Thanks! Fixed it already

File: src/components/InstancesList/InstancesList.js (hunk 2 above)
Comment (UladzislauKutarkin) [reply src/components/InstancesList/InstancesList.js]: Thanks! Fixed.

**Pull request merged**